Lib/inspect.py

```diff
@@ -173,25 +173,35 @@ def isgeneratorfunction(object):
 
     Generator function objects provide the same attributes as functions.
     See help(isfunction) for a list of attributes."""
-    return bool((isfunction(object) or ismethod(object)) and
+    is_generator_function = bool((isfunction(object) or ismethod(object)) and
                 object.__code__.co_flags & CO_GENERATOR)
+    is_partial_generator = bool((isinstance(object, functools.partial) and
+                object.func.__code__.co_flags & CO_GENERATOR))
+    return is_generator_function or is_partial_generator
```

Comment:
Wouldn't it be better to do this:

```python
while isinstance(object, functools.partial):
    object = object.func
```

? The benefit would be support of nested partials and the code will probably look less ugly.

Reply:
Also, while you're here, please rename `object` to `obj` or `o`.